src/Features/Core/Portable/MoveToNamespace/MoveToNamespaceAnalysisResult.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.MoveToNamespace
+{
+    internal partial class MoveToNamespaceAnalysisResult
+    {
+        public static MoveToNamespaceAnalysisResult Invalid = new MoveToNamespaceAnalysisResult();
+
+        public bool CanPerform { get; }
+        public Document Document { get; }
+        public SyntaxNode Container { get; }
+        public string OriginalNamespace { get; }
+        public ContainerType Type { get; }
```

Comment:
The name "Type" is confusing since it often refers to the programming languages concept. I'd prefer `ContainerType` personally.

Reply:
I'll change to `Container`, and change existing `Container` to `SyntaxNode`